chia/pools/pool_puzzles.py

```diff
@@ -0,0 +1,336 @@
+    elif is_pool_waitingroom_inner_puzzle(inner_puzzle):
+        # inner sol is (spend_type, destination_puzhash, pool_reward_amount, pool_reward_height, extra_data)
+        inner_sol = Program.to([0, 0, reward_amount, height, 0])
+    else:
+        raise ValueError
+    # full sol = (parent_info, my_amount, inner_solution)
+    coin: Optional[Coin] = get_most_recent_singleton_coin_from_coin_solution(last_coin_solution)
+    assert coin is not None
+
+    if coin.parent_coin_info == launcher_coin.name():
+        parent_info: Program = Program.to([launcher_coin.parent_coin_info, launcher_coin.amount])
+    else:
+        p = Program.from_bytes(bytes(last_coin_solution.puzzle_reveal))
+        last_coin_solution_inner_puzzle: Optional[Program] = get_inner_puzzle_from_puzzle(p)
+        assert last_coin_solution_inner_puzzle is not None
+        parent_info = Program.to(
+            [
+                last_coin_solution.coin.parent_coin_info,
+                last_coin_solution_inner_puzzle.get_tree_hash(),
+                last_coin_solution.coin.amount,
+            ]
+        )
+    full_solution: SerializedProgram = SerializedProgram.from_program(
+        Program.to([parent_info, last_coin_solution.coin.amount, inner_sol])
+    )
+    full_puzzle: SerializedProgram = SerializedProgram.from_program(
+        create_full_puzzle(inner_puzzle, launcher_coin.name())
+    )
+
+    reward_parent: bytes32 = pool_parent_id(height, genesis_challenge)
+    p2_singleton_puzzle: SerializedProgram = SerializedProgram.from_program(
+        create_p2_singleton_puzzle(SINGLETON_MOD_HASH, launcher_coin.name())
+    )
+    reward_coin: Coin = Coin(reward_parent, p2_singleton_puzzle.get_tree_hash(), reward_amount)
+    p2_singleton_solution: SerializedProgram = SerializedProgram.from_program(
+        Program.to([inner_puzzle.get_tree_hash(), reward_coin.name()])
+    )
+    assert p2_singleton_puzzle.get_tree_hash() == reward_coin.puzzle_hash
+    assert full_puzzle.get_tree_hash() == coin.puzzle_hash
+    if get_inner_puzzle_from_puzzle(Program.from_bytes(bytes(full_puzzle))) is None:
+        assert get_inner_puzzle_from_puzzle(Program.from_bytes(bytes(full_puzzle))) is not None
+
+    coin_solutions = [
+        CoinSolution(coin, full_puzzle, full_solution),
+        CoinSolution(reward_coin, p2_singleton_puzzle, p2_singleton_solution),
+    ]
+    return coin_solutions
+
+
+def get_most_recent_singleton_coin_from_coin_solution(coin_sol: CoinSolution) -> Optional[Coin]:
+    additions: List[Coin] = coin_sol.additions()
+    for coin in additions:
+        if coin.amount % 2 == 1:
+            return coin
+    return None
+
+
+def get_pubkey_from_member_inner_puzzle(inner_puzzle: Program) -> G1Element:
+    args = uncurry_pool_member_inner_puzzle(inner_puzzle)
+    if args is not None:
+        inner_f, target_puzzle_hash, p2_singleton_hash, pubkey_program, pool_reward_prefix, escape_puzzlehash = args
```

Comment:
You could add a `_` prefix before the unused variables, as in `_inner_f`. I was confused for a moment.